goflags.go

```diff
@@ -351,26 +382,71 @@ func (flagSet *FlagSet) usageFunc() {
 
 	writer := tabwriter.NewWriter(cliOutput, 0, 0, 1, ' ', 0)
 
-	flagSet.flagKeys.forEach(func(key string, data *flagData) {
-		currentFlag := flag.CommandLine.Lookup(key)
+	if len(flagSet.groups) > 0 {
+		var otherOptions []string
 
-		dataHash := data.Hash()
-		if _, ok := hashes[dataHash]; ok {
-			return // Don't print the value if printed previously
+		for _, group := range flagSet.groups {
+			fmt.Fprintf(cliOutput, "%s:\n", strings.ToUpper(group.description))
+
+			flagSet.flagKeys.forEach(func(key string, data *FlagData) {
+				if data.group == "" {
+					dataHash := data.Hash()
+					if _, ok := hashes[dataHash]; ok {
+						return // Don't print the value if printed previously
+					}
+					hashes[dataHash] = struct{}{}
+
+					currentFlag := flag.CommandLine.Lookup(key)
+					otherOptions = append(otherOptions, createUsageString(data, currentFlag))
+					return
+				}
+				// Ignore the flag if it's not in our intended group
+				if !strings.EqualFold(data.group, group.name) {
+					return
+				}
+				dataHash := data.Hash()
+				if _, ok := hashes[dataHash]; ok {
+					return // Don't print the value if printed previously
+				}
+				hashes[dataHash] = struct{}{}
+
+				currentFlag := flag.CommandLine.Lookup(key)
+				result := createUsageString(data, currentFlag)
+				fmt.Fprint(writer, result, "\n")
+			})
+			writer.Flush()
+			fmt.Printf("\n")
 		}
-		hashes[dataHash] = struct{}{}
+		if len(otherOptions) > 0 {
+			fmt.Fprintf(cliOutput, "%s:\n", strings.ToUpper(OtherOptionsGroupName))
```

Comment:
Wouldn't it make more sense to save the "OtherOptionsGroupName" directly with uppercase instead, or else you want to be uniform, then you should rather create a transformation method which would be used for all group names. This way if you decide to change the naming style of something, you'd need to modify it only in one place.

Reply:
Adding this to FlagSet itself as an option that can be changed